### PR TITLE
Numenta Dataset

### PR DESCRIPTION
**Pull request by ghost**
Let me know how it looks. Thanks Tim

### CODE REVIEW

**Comment by timmolter** [comment]
add the module to the parent pom please.

**Comment by timmolter** [comment]
use org.knowm as all package names please

**Comment by timmolter** [comment]
Please create this project like the other ones where it's made for both an in-memory HSQLDB database and optionally a MySQL database.

File: datasets-nab/pom.xml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	
+		<dependency>
+			<groupId>com.xeiam.datasets</groupId>
+			<artifactId>datasets-common</artifactId>
+			<version>2.1.0-SNAPSHOT</version>
+		</dependency>
+	
+		<dependency>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-core</artifactId>
```
Comment (timmolter) [inline datasets-nab/pom.xml]: define dependency versions in the parent pom please.

File: datasets-nab/pom.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  <groupId>com.xeiam.datasets.nab</groupId>
+  <artifactId>datasets-nab</artifactId>
+  <version>1.0-SNAPSHOT</version>
+  <name>datasets-nab</name>
+  <url>http://maven.apache.org</url>
+  
+
+	<dependencies>
+		<dependency>
+			<groupId>junit</groupId>
```
Comment (timmolter) [inline datasets-nab/pom.xml]: junit should already be a dependency as defined in the parent pom, so you don't need this here.

File: datasets-nab/pom.xml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+			<artifactId>junit</artifactId>
```
Comment (timmolter) [inline datasets-nab/pom.xml]: Also junit should always have `test` scope.

**Comment by ghost** [comment]
changes up

**Comment by JRSteeves** [comment]
I haven't been using this tool to see your comments. This really really really helps.

**Comment by JRSteeves** [comment]
Why did we switch to GitLab?

**Comment by timmolter** [comment]
Try to see what the merge conflicts are, OK. No idea what's up.

**Comment by ghost** [comment]
Hmm I'm not sure where these merge conflicts are coming from. In my repo everything is without conflict.

**Comment by JRSteeves** [comment]
oh I see.

File: datasets-numenta/src/main/java/org/knowm/datasets/numenta/NumentaDAO.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import java.util.List;
+
+import org.knowm.datasets.common.business.DatasetsDAO;
+import org.knowm.yank.Yank;
+
+public class NumentaDAO extends DatasetsDAO {
+
+  public static void init(String dataFilesDir) {
+
+    // / Needs to point to Google drive
```
Comment (timmolter) [inline datasets-numenta/src/main/java/org/knowm/datasets/numenta/NumentaDAO.java]: you can delete this init method, right? I can add it when/if I actually upload the files.

File: datasets-numenta/src/test/java/org/knowm/datasets/numenta/unit/TestNumentaDAO.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+import org.junit.Test;
+import org.knowm.datasets.numenta.NumentaDAO;
+import org.knowm.datasets.numenta.SeriesPoint;
+import org.knowm.yank.PropertiesUtils;
+import org.knowm.yank.Yank;
+
+@Ignore
+public class TestNumentaDAO {
+
+  public static void main(String[] args) {
```
Comment (timmolter) [inline datasets-numenta/src/test/java/org/knowm/datasets/numenta/unit/TestNumentaDAO.java]: No Main method in unit test classes usually. Is this necessary?

File: pom.xml [hunk 6]
```diff
@@ -88,6 +89,26 @@
...
+				<artifactId>jackson-core</artifactId>
+				<version>2.7.0</version>
+				</dependency>
+			<dependency>
+				<groupId>org.codehaus.jackson</groupId>
+				<artifactId>jackson-mapper-asl</artifactId>
+				<version>1.9.13</version>
+			</dependency>
+			<dependency>
+				<groupId>mysql</groupId>
```
Comment (timmolter) [inline pom.xml]: remove this dep.

File: datasets-numenta/src/test/java/org/knowm/datasets/numenta/bootstrap/RawData2DB.java [hunk 7]
```diff
@@ -0,0 +1,150 @@
+/**
```
Comment (timmolter) [inline datasets-numenta/src/test/java/org/knowm/datasets/numenta/bootstrap/RawData2DB.java]: remove these copyright headers

File: datasets-numenta/src/main/java/org/knowm/datasets/numenta/NumentaDAO.java [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+        point.getValue(),
+        point.getLabel()
+        // @formatter:on
+        };
+    String INSERT = "INSERT INTO NUMENTA (id, series, timestamp, value, label) VALUES (?, ?, ?, ?, ?)";
+    Yank.execute(INSERT, params);
+  }
+
+  public static List<SeriesPoint> selectSeries(String series) {
+
```
Comment (timmolter) [inline datasets-numenta/src/main/java/org/knowm/datasets/numenta/NumentaDAO.java:38]: You need to sort by timestamp

File: datasets-numenta/src/main/java/org/knowm/datasets/numenta/NumentaDAO.java [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+    List<SeriesPoint> uniqueSeries = Yank.queryBeanList(SELECT_ALL_SQL, SeriesPoint.class, null);
+    List<String> names = new ArrayList<String>();
+    for (SeriesPoint p : uniqueSeries) {
+      names.add(p.getSeries());
+    }
+    return names;
+  }
+
+  public static List<SeriesPoint> selectAll() {
+
```
Comment (timmolter) [inline datasets-numenta/src/main/java/org/knowm/datasets/numenta/NumentaDAO.java]: sort

Comment (JRSteeves) [inline datasets-numenta/src/main/java/org/knowm/datasets/numenta/NumentaDAO.java]: They are sorted going in. I have a test which check it.

File: datasets-numenta/src/main/java/org/knowm/datasets/numenta/NumentaDAO.java (hunk 8 above)
Comment (ghost) [inline datasets-numenta/src/main/java/org/knowm/datasets/numenta/NumentaDAO.java:38]: soft? oh sort

Comment (ghost) [inline datasets-numenta/src/main/java/org/knowm/datasets/numenta/NumentaDAO.java:38]: they are sorted going in. Needed?

File: datasets-numenta/src/main/java/org/knowm/datasets/numenta/NumentaDAO.java (hunk 9 above)
Comment (timmolter) [inline datasets-numenta/src/main/java/org/knowm/datasets/numenta/NumentaDAO.java]: yes, do sort.

File: datasets-numenta/src/main/java/org/knowm/datasets/numenta/NumentaDAO.java [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+    Object[] params = new Object[] { series };
+    String SELECT_ALL_SQL = "SELECT * FROM NUMENTA WHERE series = ?";
+    return Yank.queryBeanList(SELECT_ALL_SQL, SeriesPoint.class, params);
+  }
+
+  public static List<String> selectSeriesNames() {
+
+    String SELECT_ALL_SQL = "SELECT DISTINCT SERIES FROM NUMENTA";
+    List<SeriesPoint> uniqueSeries = Yank.queryBeanList(SELECT_ALL_SQL, SeriesPoint.class, null);
```
Comment (timmolter) [inline datasets-numenta/src/main/java/org/knowm/datasets/numenta/NumentaDAO.java]: You can do this more efficiently by using a different Yank method and getting the String list directly

File: datasets-numenta/src/main/java/org/knowm/datasets/numenta/NumentaDAO.java (hunk 8 above)
Comment (timmolter) [inline datasets-numenta/src/main/java/org/knowm/datasets/numenta/NumentaDAO.java:38]: Yes, do sort here.

**Pull request merged**